### PR TITLE
Security mitigation: Update the doc for suggested operator installation namespace and use custom image for RHDH RHDH

### PR DESCRIPTION
**Pull request by jianrongzhang89**

### CODE REVIEW

File: installation/proc-install-rhdh-operator-admin.adoc [hunk 1]
```diff
@@ -40,14 +40,29 @@ curl -L https://gist.githubusercontent.com/nickboldt/d54ba9d7264a5480d14eea6c3bd
...
+. In the *Administrator* perspective of the OpenShift web console, go to *Operators* → *OperatorHub*, search for Red Hat Developer Hub, and install the Red Hat Developer Hub Operator. For more info, see link:https://docs.openshift.com/container-platform/4.14/operators/admin/olm-adding-operators-to-cluster.html#olm-installing-from-operatorhub-using-web-console_olm-adding-operators-to-a-cluster[Installing from OperatorHub using the web console]. It is recommended to deploy the Backstage Operator in a dedicated default namespace `rhdh-operator`. The cluster administrator can restrict other users the access to the operator resources through RoleBindings or ClusterRoleBindings. You can choose to deploy the operator in the `openshift-operators` namespace instead. However, you should keep in mind that the Backstage Operator shares the namespace with other operators and therefore any users who can create workloads in that namespace can get their privileges escalated from all operators' service accounts.
```
Comment (nickboldt) [inline installation/proc-install-rhdh-operator-admin.adoc:43]: if you create the `operator` in `openshift-operators` (per the recommended guidelines for an `AllNamespaces` subscription) there's no need to create the `backstage` CRs in the same namespace. 

```
The namespace to which you subscribe the Operator must have an Operator group that matches the install mode of the Operator, either the AllNamespaces or SingleNamespace mode. If the Operator you intend to install uses the AllNamespaces mode, then the openshift-operators namespace already has an appropriate Operator group in place.

However, if the Operator uses the SingleNamespace mode and you do not already have an appropriate Operator group in place, you must create one.
```

https://docs.openshift.com/container-platform/4.14/operators/user/olm-installing-operators-in-namespace.html

File: installation/proc-install-rhdh-operator.adoc [hunk 2]
```diff
@@ -46,7 +46,22 @@ This section describes the installation procedure for installing {product} on yo
...
+[source,yaml]
+----
+spec:
+  application:
+    image: <your custom image>
+----
+
+[WARNING]
+====
+Installing a {product} application with a custom {product} image is at your own risk. It is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance. Red Hat only supports the images shipped within the {product} Operator.
```
Comment (nickboldt) [inline installation/proc-install-rhdh-operator.adoc:66]:
```suggestion
Installing a {product} application with a custom {product} image is at your own risk. It is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance. Red Hat only supports the images shipped with the {product} Operator which have been published to the Red Hat Ecosystem Catalog at https://catalog.redhat.com/software/containers/search?q=rhdh.
```

File: installation/proc-install-rhdh-operator-admin.adoc [hunk 3]
```diff
@@ -40,14 +40,29 @@ curl -L https://gist.githubusercontent.com/nickboldt/d54ba9d7264a5480d14eea6c3bd
...
+[source,yaml]
+----
+spec:
+  application:
+    image: <your custom image>
+----
+
+[WARNING]
+====
+Installing a {product} application with a custom {product} image is at your own risk. It is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance. Red Hat only supports the images shipped within the {product} Operator.
```
Comment (nickboldt) [inline installation/proc-install-rhdh-operator-admin.adoc:67]:
```suggestion
Installing a {product} application with a custom {product} image is at your own risk. It is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance. Red Hat only supports the images shipped with the {product} Operator which have been published to the Red Hat Ecosystem Catalog at https://catalog.redhat.com/software/containers/search?q=rhdh.
```

**Review by nickboldt — CHANGES_REQUESTED** [review]

File: installation/proc-install-rhdh-operator-admin.adoc (hunk 1 above)
Comment (jianrongzhang89) [reply installation/proc-install-rhdh-operator-admin.adoc:43]: @nickboldt did you mean backstage Operator Group instead? The downside of deploying the operator in openshift-operators is it shares the namespace with other operators. So if a user ever gets the permission to create workloads in that namespace, he would be be able to get his privileges escalated from all the service accounts in that namespace. Deploying the operator in a separate namespace minimizes such privilege escalation.

**Comment by jianrongzhang89** [comment]
This PR is no longer as we decided to use the google doc to keep track of the doc changes.

**Pull request closed without merging**